pkgs/tools/virtualization/mkosi/default.nix

```diff
@@ -38,34 +40,51 @@ let
     withKernelInstall = true;
   };
 
-  python3pefile = python3.withPackages (ps: with ps; [
-    pefile
-  ]);
+  python3pefile = python3.withPackages (
+    ps: with ps; [
+      pefile
+    ]
+  );
 in
 buildPythonApplication rec {
   pname = "mkosi";
-  version = "22";
+  version = "24.3-unstable-2024-08-28";
   format = "pyproject";
 
-  outputs = [ "out" "man" ];
+  outputs = [
+    "out"
+    "man"
+  ];
 
   src = fetchFromGitHub {
     owner = "systemd";
     repo = "mkosi";
-    rev = "v${version}";
-    hash = "sha256-Zom1GlyhqgpTKfjcBOUEJMlubSn+TQsk97js1/UfDHY=";
+    rev = "8c2f828701a1bdb3dc9b80d6f2ab979f0430a6b8";
+    hash = "sha256-rO/4ki2nAJQN2slmYuHKESGBBDMXC/ikGf6dMDcKFr4=";
   };
 
-  # Fix ctypes finding library
-  # https://github.com/NixOS/nixpkgs/issues/7307
-  postPatch = lib.optionalString stdenv.isLinux ''
-    substituteInPlace mkosi/user.py \
-      --replace-fail 'ctypes.util.find_library("c")' "'${stdenv.cc.libc}/lib/libc.so.6'"
-    substituteInPlace mkosi/__init__.py \
-      --replace-fail '/usr/lib/systemd/ukify' "${systemdForMkosi}/lib/systemd/ukify"
-  '' + lib.optionalString withQemu ''
-    substituteInPlace mkosi/qemu.py \
-      --replace-fail "usr/share/qemu/firmware" "${qemu}/share/qemu/firmware"
+  patches =
+    [
+      (replaceVars ./0001-Use-wrapped-binaries-instead-of-Python-interpreter.patch {
+        UKIFY = "${systemdForMkosi}/lib/systemd/ukify";
+        PYTHON_PEFILE = "${python3pefile}/bin/python3.12";
+        MKOSI_SANDBOX = "~MKOSI_SANDBOX~"; # to satisfy replaceVars, will be replaced in postPatch
```

Comment:
This is a funny trick, kind of a hack, but it absolutely works. I was thinking of introducing a variant of `replaceVars` (maybe `replaceVars.withoutCheck`) that skipped the check phase so these hacks wouldn't be necessary.

If `buildPythonApplication` supported `finalAttrs`, would this be `"${finalAttrs.finalPackage.outPath}/bin/mkosi-sandbox"`?

Reply:
I made a PR to implement this behavior.